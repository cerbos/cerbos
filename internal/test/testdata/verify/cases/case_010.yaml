description: Duplicate test (#1411)
