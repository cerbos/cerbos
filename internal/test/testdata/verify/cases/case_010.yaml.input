-- testdata/principals.yaml --
---
principals:
  john:
    policyVersion: '20210210'
    id: john
    roles:
      - employee
    attr:
      department: marketing
      geography: GB
      team: design

  alicia:
    id: alicia
    policyVersion: '20210210'
    roles:
      - employee
    attr:
      department: marketing
      geography: GB
      team: design

-- testdata/resources.yaml --
---
resources:
  john_leave_request:
    policyVersion: '20210210'
    kind: leave_request
    id: XX125
    attr: &attr
      department: marketing
      geography: GB
      id: XX125
      owner: john
      team: design

-- suite_test.yaml --
---
name: TestSuite
description: Tests
tests:
  - name: Test 1
    input:
      principals:
        - john
      resources:
        - john_leave_request
      actions:
        - view:public
    expected:
      - principal: john
        resource: john_leave_request
        actions:
          view:public: EFFECT_ALLOW

  - name: Test 1
    input:
      principals:
        - alicia
      resources:
        - john_leave_request
      actions:
        - view:public
    expected:
      - principal: john
        resource: john_leave_request
        actions:
          view:public: EFFECT_ALLOW

      - principal: alicia
        resource: john_leave_request
        actions:
          view:public: EFFECT_DENY
