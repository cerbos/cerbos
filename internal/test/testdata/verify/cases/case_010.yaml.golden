{
  "suites": [
    {
      "file": "suite_test.yaml",
      "name": "TestSuite",
      "summary": {
        "overallResult": "RESULT_ERRORED"
      },
      "error": "Invalid test suite: another test named \"Test 1\" already exists"
    }
  ],
  "summary": {
    "overallResult": "RESULT_ERRORED"
  }
}